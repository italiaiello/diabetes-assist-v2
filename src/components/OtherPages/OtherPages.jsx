import React from 'react';
import Cybel from '../Cybel/Cybel'
import Emergency from '../Emergency/Emergency'
import Health from '../Health/Health'
import MealPlanner from '../MealPlanner/MealPlanner'
import Appointment from '../Appointment/Appointment'
import SignIn from '../SignIn/SignIn'
import Register from '../Register/Register'
import { useDataFetch } from '../../hooks/DisplayRecipes'

const OtherPages = props => {

    let route = props.route
    const [isLoading, recipeData] = useDataFetch('https://api.spoonacular.com/recipes/search?apiKey=d464b770a6c2452cb7d56fc6ccea9eb5', props.fetchData)

    return (
        <div>
            {

            isLoading && route === 'meal' ? 
            <h1>Fetching Recipe Data</h1>
            :
            (
                isLoading ?
                <h1>Loading...</h1>
                :
                (
                    route === 'signin' ?
                    <SignIn onRouteChange={props.onRouteChange} loadUser={props.loadUser} />
                    :
                    (
                        route === 'register' ?
                        <Register onRouteChange={props.onRouteChange} loadUser={props.loadUser} />
                        :
                        (
                            route === 'health' ?
                            <Health onRouteChange={props.onRouteChange}/>
                            :
                            (
                                route === 'cybel' ?
                                <Cybel onRouteChange={props.onRouteChange} />
                                :
                                (
                                    route === 'meal' || route === 'mealSelect' || route === 'recipe'
                                    ?
                                    <MealPlanner recipeData={recipeData}
                                                onRouteChange={props.onRouteChange}
                                                route={route}
                                                userEmail={props.userEmail}
                                                
                                    />
                                    :
                                    (
                                        route === 'appointment' || route === 'appointmentTime' || 
                                        route === 'confirmBooking' 
                                        ?
                                        <Appointment route={route} 
                                                        onRouteChange={props.onRouteChange}
                                                            userEmail={props.userEmail} />
                                        :
                                        (
                                            route === 'Emergency' &&
                                            <Emergency onRouteChange={props.onRouteChange} />
                                        )                               
                                        
                                
                                        
                                    )
                                )
                            )
                        )
                    )
                )
            )
        }
        </div>
    )
}

export default OtherPages;