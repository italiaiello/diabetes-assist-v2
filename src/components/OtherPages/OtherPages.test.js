import { shallow } from 'enzyme';
import React  from 'react';
import OtherPages from './OtherPages';

describe('OtherPages.js', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<OtherPages />)
  })

  it('expects to render TodoList component', () => {
    expect(wrapper).toMatchSnapshot()
  })

})