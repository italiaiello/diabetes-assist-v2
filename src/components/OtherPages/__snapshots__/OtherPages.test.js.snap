// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OtherPages.js expects to render TodoList component 1`] = `ShallowWrapper {}`;
